### ISSUE TITLE
on 1.8+ enabling https server breaks http server with busy error

### ISSUE THREAD

**Issue by spoonincode** [EOSIO/eos#10695]
This defect seems to be in all nodeos 1.8+

Launch nodeos with simply
```
nodeos --https-server-address 127.0.0.1:5555 --plugin eosio::chain_api_plugin
```

Trying to access the http server fails
```
$ curl http://127.0.0.1:8888/v1/chain/get_info
{"code":429,"message":"Busy","error":{"code":429,"name":"Busy","what":"Too many bytes in flight: 828. Try again later.","details":[]}}
```

**Comment by praphael** [comment]
It should probably be an error to use --https-server-address  without valid  --https-certificate-chain-file and  --https-private-key-file

**Comment by heifner** [comment]
Since this has been broken since 1.8+ and not reported. I suggest we remove support for https. Obviously people are providing https support via proxies. I see no reason to provide it in nodeos/rodeos/keosd.

**Comment by praphael** [comment]
It appears the problem was caused by the order of initialization in http_plugin::plugin_initialize(), and an early return, When it fails to initialize the https endpoint, due to missing SSL certificate or private key files, it will return early and not initialize several options, including max_bytes_in_flight,  which defaulted to 0.   The plugin would then continue to use non-SSL HTTP on the default port, which also happens to be the port specified by the user.  

Should be fixed as part of this PR
https://github.com/EOSIO/eos/pull/10689

Failure to initialize SSL properly should be error which causes rodeos initialization to fail - it should not just continue to use non-SSL on the same port.  Otherwise users would think that SSL is initialized when it is not, unless they pay attention to the error logs which can run by fast.

**Comment by heifner** [comment]
If we are going to fix it, then it should be fixed in 2.0.x, 2.1.x, 2.2.x, and in #10689. I think we should remove it.

**Comment by aclark-b1** [comment]
Chatted with @heifner a bit more and it sounds like we need this to support http_client for Zipkin, so we should go ahead and fix it rather than removing it.

**Comment by praphael** [comment]
Fixed in 2.0, 2.1, and 2.2.
